bw_timex/timex_lca.py

```diff
@@ -1053,35 +1055,89 @@ def create_node_id_collection_dict(self) -> None:
 
         self.node_id_collection_dict["first_level_background_node_ids_static"] = (
             first_level_background_node_ids_static
         )
-        self.node_id_collection_dict["first_level_background_node_ids_all"] = (
-            first_level_background_node_ids_all
-        )
+   
+
+    def add_interdatabase_activity_mapping_from_timeline(self) -> None:
+        """
+        Fills the interdatabase_activity_mapping, which is a SetList of the matching processes across background databases
+        in the format of {(id, database_name_1), (id, database_name_2)} with only those activities and background databases that are actually mapped in the timeline.
+        Also adds the ids to the node_id_collection_dict["first_level_background_node_ids_interpolated"].
+        This avoids unneccessary peewee calls.
+
+        Parameters
+        ----------
+        None
+
+
+        Returns
+        -------
+        None, but adds the ids of producers in other background databases (only those interpolated to in the timeline) to the `interdatabase_activity_mapping` and `node_id_collection["first_level_background_node_ids_interpolated"]`.
+        """
+        if not hasattr(self, "timeline"):
+            warnings.warn("Timeline not yet built. Call TimexLCA.build_timeline() first.")
+            return
+                          
+        self.interdatabase_activity_mapping = SetList()
+        first_level_background_node_ids_interpolated = set()
+        unique_producer_background_db_combos = {}
+
+        # get unique combos of producers and background dbs from timeline
+        for _, row in self.timeline.iterrows():
+            if row["interpolation_weights"] is not None: # "None" corresponds to exchanges linked within the foreground -> skipped
+                
+                if row['producer'] not in unique_producer_background_db_combos.keys():
+                    unique_producer_background_db_combos[row['producer']] = set()
+                unique_producer_background_db_combos[row['producer']].update(row["interpolation_weights"].keys())
+
+        # fill interdatabase_activity_mapping with unique combos
+        for original_producer_id, background_databases in unique_producer_background_db_combos.items():
+            original_database = list(self.node_id_collection_dict["demand_dependent_background_database_names"])[0] #what if more than one orignal bg database is linked?
+            orig_act = bd.get_node(id=original_producer_id)
+            act_set = set()
+
+            for background_db in background_databases:
+                try:
+                    other_node = bd.get_node(
+                        **{
+                            "database": background_db,
+                            "name": orig_act["name"],
+                            "product": orig_act["reference product"],
+                            "location": orig_act["location"],
+                        }
```

Comment:
I think while we're at it, we might as well refactor this into a separate function, e.g. "find_matching_node(original_node, other_database)" or similar. This would be more flexible looking forwards, maybe there will be other matching strategies at some point.

Reply:
good idea, implemented in df29e31a0980eb5fb2191cf8f64a5ff91a8ac0da